configuration/impl/src/main/java/org/jboss/forge/addon/configuration/proxy/ForgeProxySelector.java

```diff
@@ -81,12 +84,20 @@ private boolean isProxyAvailable(final URI uri)
       boolean isValidProtocol =
                "http".equalsIgnoreCase(protocol) || "https".equalsIgnoreCase(protocol);
 
-      boolean isHostExcluded =
-               proxySettings.getNonProxyHosts() != null
-                        && proxySettings.getNonProxyHosts().contains(host);
+      boolean isHostExcluded = proxySettings.getNonProxyHosts() != null && matches(host);
       return isValidProtocol && !isHostExcluded;
    }
 
+   private boolean matches(String target)
```

Comment:
Maybe this method could receive the list of non-proxy hosts instead (and the null check could happen inside this method as well)

Reply:
the uri target is the variable, the nonProxyHosts are the known.
Is this relevant, because your proposal below is _matchesHosts(String target)_ also?

Reply:
It's an instance method, so I don't think it's relevant